packages/server/graphql/private/mutations/verifyDomain.ts

```diff
@@ -0,0 +1,28 @@
+import getRethink from '../../../database/rethinkDriver'
+import {MutationResolvers} from '../resolverTypes'
+import {normalizeSlugName} from './helpers/SAMLHelpers'
+
+const verifyDomain: MutationResolvers['verifyDomain'] = async (_source, {slug, domains, orgId}) => {
+  const r = await getRethink()
+  const normalizedDomains = domains.map((domain) => domain.toLowerCase())
+
+  // VALIDATION
+  const slugName = normalizeSlugName(slug)
+  if (slugName instanceof Error) return {error: {message: slugName.message}}
+
+  const organizationSAMLExist = await r.table('SAML').get(slugName).run()
```

Comment:
We can check if record exists without fetching the whole data by  ` .count().eq(1)`